app/scripts/controllers/network/network-controller.js

```diff
@@ -268,14 +269,20 @@ export default class NetworkController extends EventEmitter {
       NETWORK_TYPES.RPC,
       `NetworkController - cannot call "setProviderType" with type "${NETWORK_TYPES.RPC}". Use "setActiveNetwork"`,
     );
-    assert.ok(
-      INFURA_PROVIDER_TYPES.includes(type),
-      `Unknown Infura provider type "${type}".`,
-    );
+    if (type !== NETWORK_TYPES.LINEA_TESTNET) {
```

Comment:
why is this if block being added?

Reply:
hmm... I am reading this comment https://github.com/MetaMask/metamask-extension/pull/17875#discussion_r1122182027

Reply:
in any case, we shouldn't create a new type for this network. It should either be a `INFURA_PROVIDER_TYPES` or a  `NETWORK_TYPES.RPC`

I would suggest we treat this as a `NETWORK_TYPES.RPC`

Reply:
and as opposed to modifying the network controller, we can just modify `ui/components/app/dropdowns/network-dropdown.js`

treating this network in a similar way to how localhost is treated, within the render function of that component

Reply:
it can still be in the network dropdown, we just need to place it there within `ui/components/app/dropdowns/network-dropdown.js`

Reply:
similarly to how localhost is added to the network dropdown